client/client.go

```diff
@@ -904,3 +904,32 @@ func (r *NotaryRepository) rootFileKeyChange(role, action string, key data.Publi
 	}
 	return nil
 }
+
+// DeleteTrustData removes the trust data stored for this repo in the TUF cache on the client side,
+// if clearCertificate is set, it will also delete the certificate data for this GUN
+func (r *NotaryRepository) DeleteTrustData() error {
+	// Clear TUF files and cache
+	if err := r.fileStore.RemoveAll(); err != nil {
+		return fmt.Errorf("error clearing TUF repo data: %v", err)
+	}
+	r.tufRepo = tuf.NewRepo(nil, nil)
+	// Clear certificates
+	certStore := r.CertManager.TrustedCertificateStore()
+	certificates, err := certStore.GetCertificatesByCN(r.gun)
+	if err != nil {
+		return fmt.Errorf("error retrieving certificates for %s: %v", r.gun, err)
+	}
+	for _, cert := range certificates {
+		if err := certStore.RemoveCert(cert); err != nil {
+			return fmt.Errorf("error removing certificate: %v: %v", cert, err)
+		}
+	}
+	// Clear keys
+	repoKeys := r.CryptoService.ListAllKeys()
```

Comment:
If we're moving towards a flat key store, I wouldn't bother deleting the old keys. It doesn't affect what we trust, only what we can sign. I also worry that we may delete keys we shouldn't be deleting. I don't think that's happening here but deleting a key you need leaves you in a much worse state than having an obsolete key taking up disk space.